### PR TITLE
[RPA-606] Searching processes by tag

### PR DESCRIPTION
**Pull request by Aeri4a**
## Description

Zrzut ekranu 2023-09-19 104815

Zrzut ekranu 2023-09-19 104802

[+] Additionally fixed bug with not showing processes by search after edit action.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

---

I agree to the terms of the RunBotics Contributor License Agreement.

### CODE REVIEW

File: runbotics-orchestrator/src/main/java/com/runbotics/service/criteria/ProcessCriteria.java [hunk 1]
```diff
@@ -61,6 +63,7 @@ public ProcessCriteria(ProcessCriteria other) {
         this.createdById = other.createdById == null ? null : other.createdById.copy();
         this.createdByName = other.createdByName == null ? null : other.createdByName.copy();
         this.botCollectionName = other.botCollectionName == null ? null : other.botCollectionName.copy();
+        this.tagName = other.tagName == null ? null : other.tagName.copy();
```
Comment (marcin-depcik) [inline runbotics-orchestrator/src/main/java/com/runbotics/service/criteria/ProcessCriteria.java:66]: perhaps it is better to assign the value of other.tagName to preserve the source of truth, such as:
this.tagName = other.tagName == null ? other.tagName : other.tagName.copy();

File: runbotics-orchestrator/src/main/java/com/runbotics/service/ProcessQueryService.java [hunk 2]
```diff
@@ -122,6 +131,30 @@ public Page<ProcessDTO> findByCriteria(ProcessCriteria criteria, Pageable page)
             .map(processMapper::toDto);
     }
 
+    @Transactional(readOnly = true)
+    public Page<ProcessDTO> findBySearchField(ProcessCriteria criteria, Pageable page, User user) {
+        log.debug("Request to get processes by page using search field: {}", criteria, page);
```
Comment (marcin-depcik) [inline runbotics-orchestrator/src/main/java/com/runbotics/service/ProcessQueryService.java]: I think you missed the placeholder for 'page':
log.debug("Request to get processes by page using search field: {} {}", criteria, page);

File: runbotics/runbotics-orchestrator-ui/src/src-app/components/Tile/ProcessTile/ProcessTileTagList/ProcessTileTagList.tsx [hunk 3]
```diff
@@ -64,15 +67,25 @@ const ProcessTileTagList: FC<ProcessTileTagListProps> = ({ tags }) => {
...
+                                <HighlightText
+                                    text={tag.name}
+                                    matchingText={searchValue}
+                                    matchStyle={highlightTextStyle}
+                                />
+                            }
+                            key={tag.name}
+                            size='small'
+                        />)
                 }
```
Comment (marcin-depcik) [inline runbotics/runbotics-orchestrator-ui/src/src-app/components/Tile/ProcessTile/ProcessTileTagList/ProcessTileTagList.tsx]: you can refactor this and use reduce method

`{tags
    .reduce((acc, tag) => 
      tagNameFitSearch(tag.name) 
          ? [tag, ...acc] 
          : [...acc, tag]
    , [])
    .map((tag) =>
        <Chip
            label={
                <HighlightText
                    text={tag.name}
                    matchingText={searchValue}
                    matchStyle={highlightTextStyle}
                />
            }
            key={tag.name}
            size='small'
        />)
}`

**Review by marcin-depcik — CHANGES_REQUESTED** [review]

File: runbotics/runbotics-orchestrator-ui/src/src-app/components/Tile/ProcessTile/ProcessTileTagList/ProcessTileTagList.tsx (hunk 3 above)
Comment (Aeri4a) [reply runbotics/runbotics-orchestrator-ui/src/src-app/components/Tile/ProcessTile/ProcessTileTagList/ProcessTileTagList.tsx]: Done.

File: runbotics-orchestrator/src/main/java/com/runbotics/service/ProcessQueryService.java (hunk 2 above)
Comment (Aeri4a) [reply runbotics-orchestrator/src/main/java/com/runbotics/service/ProcessQueryService.java]: You're right! I have to add one more.

File: runbotics-orchestrator/src/main/java/com/runbotics/service/criteria/ProcessCriteria.java (hunk 1 above)
Comment (Aeri4a) [reply runbotics-orchestrator/src/main/java/com/runbotics/service/criteria/ProcessCriteria.java:66]: This is clone implementation, so it has to be like the others above.

**Review by preludium — APPROVED** [review]

**Review by marcin-depcik — APPROVED** [review]

**Pull request merged**